raphtory/src/search/node_index.rs

```diff
@@ -0,0 +1,325 @@
+                TextFieldIndexing::default()
+                    .set_tokenizer(TOKENIZER)
+                    .set_index_option(IndexRecordOption::WithFreqsAndPositions),
+            ),
+        );
+        schema_builder.add_text_field(
+            fields::NODE_TYPE,
+            TextOptions::default().set_indexing_options(
+                TextFieldIndexing::default()
+                    .set_tokenizer(TOKENIZER)
+                    .set_index_option(IndexRecordOption::WithFreqsAndPositions),
+            ),
+        );
+        schema_builder
+    }
+
+    pub fn get_node_field(&self, field_name: &str) -> tantivy::Result<Field> {
+        self.index.schema().get_field(field_name)
+    }
+
+    fn create_document<'a>(
+        &self,
+        node_id: u64,
+        node_name: String,
+        node_type: ArcStr,
+    ) -> tantivy::Result<TantivyDocument> {
+        let schema = self.index.schema();
+
+        let doc = json!({
+            "node_id": node_id,
+            "node_name": node_name,
+            "node_type": node_type,
+        });
+
+        let document = TantivyDocument::parse_json(&schema, &doc.to_string())?;
+        // println!("Added node doc: {}", &document.to_json(&schema));
+
+        Ok(document)
+    }
+
+    fn collect_constant_properties<'graph, G: GraphViewOps<'graph>, GH: GraphViewOps<'graph>>(
+        &self,
+        node: &NodeView<G, GH>,
+    ) -> Vec<(ArcStr, usize, Prop)> {
+        node.properties()
+            .constant()
+            .iter()
+            .filter_map(|(k, p)| node.get_const_prop_id(k.as_ref()).map(|id| (k, id, p)))
+            .collect()
+    }
+
+    fn collect_temporal_properties<'graph, G: GraphViewOps<'graph>, GH: GraphViewOps<'graph>>(
+        &self,
+        node: &NodeView<G, GH>,
+    ) -> Vec<(i64, ArcStr, usize, Prop)> {
+        node.properties()
+            .temporal()
+            .iter()
+            .flat_map(|(key, values)| {
+                let pid = values.id;
+                values
+                    .into_iter()
+                    .map(move |(t, v)| (t, key.clone(), pid, v))
+            })
+            .collect_vec()
+    }
+
+    fn index_node<'graph, G: GraphViewOps<'graph>, GH: GraphViewOps<'graph>>(
+        &self,
+        node: NodeView<G, GH>,
+        writer: &IndexWriter,
+        const_writers: &[Option<IndexWriter>],
+        temporal_writers: &[Option<IndexWriter>],
+    ) -> tantivy::Result<()> {
+        let node_id: u64 = usize::from(node.node) as u64;
+        let node_name = node.name();
+        let node_type = node.node_type().unwrap_or_else(|| ArcStr::from(""));
+
+        let const_properties = self.collect_constant_properties(&node);
+        index_node_const_properties(
+            const_properties.iter().cloned(),
+            self.constant_property_indexes.write()?,
+            node_id,
+            const_writers,
+        )?;
+
+        let temporal_properties = self.collect_temporal_properties(&node);
```

Comment:
same here, if you drop the collect and use iter as in edge_index.rs you'll avoid some allocations